File: app/src/main/java/com/feriantes4dawin/feriavirtualmovil/ui/settings/SettingsActivity.java
package com.feriantes4dawin.feriavirtualmovil.ui.settings;

import android.os.Bundle;
import android.util.Log;

import androidx.appcompat.app.AppCompatActivity;
import androidx.lifecycle.ViewModelProvider;

import com.feriantes4dawin.feriavirtualmovil.R;
import com.feriantes4dawin.feriavirtualmovil.ui.util.FeriaVirtualConstants;

public class SettingsActivity extends AppCompatActivity {

    private SettingsViewModel viewmodel;

    @Override
    public void onCreate(Bundle savedInstanceState) {

        super.onCreate(savedInstanceState);

        setContentView(R.layout.activity_settings);

        if (savedInstanceState == null) {
            this.getSupportFragmentManager()
                .beginTransaction()
                .replace(R.id.settings2, new SettingsFragment())
                .commit();
        }
        if(getSupportActionBar() != null){
            getSupportActionBar().setDisplayHomeAsUpEnabled(true);
        }

        this.viewmodel = new ViewModelProvider(this).get(SettingsViewModel.class);
    }

    @Override
    public void onBackPressed() {
        // your code.
        finish();
    }

}

File: app/src/main/java/com/feriantes4dawin/feriavirtual/ui/settings/SettingsActivity.kt
package com.feriantes4dawin.feriavirtual.ui.settings

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import androidx.lifecycle.ViewModelProvider
import androidx.preference.PreferenceFragmentCompat
import com.feriantes4dawin.feriavirtual.R

class SettingsActivity : AppCompatActivity() {

    private lateinit var viewmodel : SettingsViewModel

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.settings_activity)
        if (savedInstanceState == null) {
            supportFragmentManager
                .beginTransaction()
                .replace(R.id.settings2, SettingsFragment())
                .commit()
        }
        supportActionBar?.setDisplayHomeAsUpEnabled(true)

        viewmodel = ViewModelProvider(this).get(SettingsViewModel::class.java)
    }

    class SettingsFragment : PreferenceFragmentCompat() {
        override fun onCreatePreferences(savedInstanceState: Bundle?, rootKey: String?) {
            setPreferencesFromResource(R.xml.root_preferences, rootKey)
        }
    }
}